### PR TITLE
Delete S4787 (Deprecated)

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Rspec update: https://github.com/SonarSource/rspec/pull/3000

=================================
Deleting S4787 as it has been deprecated.
Deprecated since:

sonar-dotnet 8.9.0.19135, released on Jun 26, 2020
SQ 8.4.0.35506 on Jul 3, 2020.

### CODE REVIEW

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
The changes to the ruleset file seem weird. The rest looks fine.

File: analyzers/its/AllSonarAnalyzerRules.ruleset [hunk 1]
```diff
@@ -2267,6 +2267,7 @@
     <Rule Id="S2252" Action="Warning" />
     <Rule Id="S2253" Action="Warning" />
     <Rule Id="S2254" Action="Warning" />
+    <Rule Id="S2255" Action="Warning" />
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply analyzers/its/AllSonarAnalyzerRules.ruleset:2270]: No - so I deleted it in the previous rule deletion S2255 but I should not have.
This is just a ruleset of continuous ids. I should not have touched it.

File: analyzers/its/SingleRule.ruleset.template [hunk 2]
```diff
@@ -14808,6 +14810,7 @@
     <Rule Id="S4786" Action="None" />
     <Rule Id="S4787" Action="None" />
     <Rule Id="S4788" Action="None" />
+    <Rule Id="S4788" Action="None" />
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply analyzers/its/SingleRule.ruleset.template:14813]: I deleted and re-added it. 
Then rebased and re-added twice probably.
Good catch thanks!

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM

**Pull request merged**